File: 05-recursive-factorial/recursive_factorial.py
def recursive_factorial(n: int) -> int:
    # Base cases
    if n == 0 or n == 1:
        return 1
    
    # Recursive case
    return n * recursive_factorial(n - 1)

def main():
    # Read input
    with open('input.txt', 'r') as f:
        n = int(f.readline())    
    # Calculate result
    result = recursive_factorial(n)
    
    # Write output
    with open('output.txt', 'w') as f:
        f.write(str(result) + '\n')

if __name__ == "__main__":
    main() 

File: 05-recursive-factorial/recursive_factorial.cpp
#include <iostream>
#include <fstream>
using namespace std;

long long recursiveFactorial(int n) {
    // Base cases
    if (n == 0 || n == 1) {
        return 1;
    }
    
    // Recursive case
    return n * recursiveFactorial(n - 1);
}

int main() {
    // Read input
    ifstream input("input.txt");
    int n;
    input >> n;
    input.close();
    
    // Calculate result
    long long result = recursiveFactorial(n);
    
    // Write output
    ofstream outFile("output.txt");
    outFile << result << endl;
    
    return 0;
} 